### DISCUSSION TITLE
Could Plots pane show all figures created using purrr::map() or similar?

### DISCUSSION THREAD

**Discussion by mpaulacaldas** [posit-dev/positron#4614] [Feedback]
When I'm doing EDA within RStudio, I often create many images at once using a combination of `ggplot2` and `purrr::map()`, which I then quickly scroll through in the Plots pane. If I try to do the same in Positron (v 2024.09.0-1), only the last image gets rendered and it's impossible to navigate to the others in the Plots pane.

<details><summary>Code example</summary>

```r
library(ggplot2)
library(purrr)

economics_list <- split(economics_long, economics_long$variable)

# In the RStudio IDE: all images get rendered to the Plots pane, and the user
# can navigate back and forth between the rendered images.
#
# In Positron: only the last image gets rendered, it's impossible to navigate
# to previous ones.
# 
economics_list |> 
  imap(
    \(df, nm) df |> 
      ggplot(aes(date, value)) +
      geom_line() +
      labs(title = nm)
      )
```

</details>
<details><summary>In RStudio (I can scroll to previous figures)</summary>

![image](https://github.com/user-attachments/assets/be6bbb3d-d4b7-4045-b4bb-bf9d1241bfe6)

</details>

<details><summary>In Positron (only last figure is available)</summary>

![image](https://github.com/user-attachments/assets/b63d4509-e3d9-49dc-b7ea-72736ecfea1f)

</details>

Would it be possible to support this sort of use-case in the Plots pane? It is a fairly common EDA pattern for me, but I suspect also for other frequent users of `purrr::map()` with `ggplot2`. It's a really useful pattern for medium-sized data sets where there are too many features/categories to plot in a single graph, but still few enough that we can manually go through each figure. There are of course work-arounds (e.g. assign the plots to an object and inspect each one, or save the images on disk), but they kind of slow down 
 the flow of exploration.

**Comment by juliasilge** [comment]
We've got this tracked in https://github.com/posit-dev/positron/issues/727 if you'd like to add any additional details. Thank you for the feedback! 🙌
